core/src/main/scala/org/broadinstitute/dsde/rawls/util/WorkspaceSupport.scala

```diff
@@ -75,31 +68,18 @@ trait WorkspaceSupport {
     accessCheck(workspace, requiredAction, ignoreLock = true) flatMap { _ => codeBlock }
 
   def requireComputePermission(workspaceName: WorkspaceName): Future[Unit] =
-    getWorkspaceContext(workspaceName).flatMap { workspace =>
-      def require(action: SamResourceAction, mkThrowable: WorkspaceName => Throwable) =
-        raiseUnlessUserHasAction(action, SamResourceTypeNames.workspace, workspace.workspaceId, ctx) {
-          mkThrowable(workspaceName)
-        }
-
-      require(SamWorkspaceActions.compute, WorkspaceAccessDeniedException.apply).recoverWith { case t: Throwable =>
-        // verify the user has `read` on the workspace to avoid exposing its existence
-        require(SamWorkspaceActions.read, NoSuchWorkspaceException.apply) *> Future.failed(t)
+    for {
+      _ <- userEnabledCheck
+      workspace <- getWorkspaceContext(workspaceName)
+      workspaceId = workspace.workspaceId
+      _ <- raiseUnlessUserHasAction(SamWorkspaceActions.compute, SamResourceTypeNames.workspace, workspaceId) {
+        WorkspaceAccessDeniedException(workspaceName)
+      }.recoverWith { case t: Throwable =>
+        raiseUnlessUserHasAction(SamWorkspaceActions.read, SamResourceTypeNames.workspace, workspaceId) {
```

Comment:
Can you add the comment below back in? I was confused about this raise until I read the comment in the previous version.

```
 // verify the user has `read` on the workspace to avoid exposing its existence